src/EnergyPlus/OutputProcessor.hh

```diff
@@ -170,375 +154,474 @@ namespace OutputProcessor {
-        }
+        std::string key = "";             // Could be blank or "*"
+        std::string name = "";         // Name of Variable
+        ReportFreq freq = ReportFreq::Hour; // Reporting Frequency
+        int SchedPtr = 0;                 // Index of the Schedule
+        std::string SchedName = "";       // Schedule Name
+        bool Used = false;                // True when this combination (key, varname, frequency) has been set
+
+        bool is_simple_string = true;     // Whether the Key potentially includes a Regular Expression pattern
+        std::shared_ptr<RE2> case_insensitive_pattern = nullptr;
     };
 
-    struct MeterArrayType
+    struct MeterPeriod
+    {
+        Real64 Value = 0.0;          // Daily Value
+        Real64 MaxVal = MaxSetValue;         // Maximum Value
+        int MaxValDate = -1;        // Date stamp of maximum
+        Real64 MinVal = MinSetValue;         // Minimum Value
+        int MinValDate = -1;        // Date stamp of minimum
+
+        bool Rpt = false;              // Report at End of Day
+        bool RptFO = false;            // Report at End of Day -- meter file only
+        int RptNum = 0;            // Report Number
+        bool accRpt = false;
+        bool accRptFO = false;
+        int accRptNum = 0;
+            
+        void resetVals() { Value = 0.0; MaxVal = MaxSetValue; MaxValDate = 0; MinVal = MinSetValue; MinValDate = 0; }
+    };
+        
+        
+    struct Meter
     {
         // Members
-        int NumOnMeters;            // Number of OnMeter Entries for variable
-        int RepVariable;            // Backwards pointer to real Variable
-        Array1D_int OnMeters;       // Forward pointer to Meter Numbers
-        int NumOnCustomMeters;      // Number of OnCustomMeter Entries for variable
-        Array1D_int OnCustomMeters; // Forward pointer to Custom Meter Numbers
-
-        // Default Constructor
-        MeterArrayType() : NumOnMeters(0), RepVariable(0), OnMeters(10, 0), NumOnCustomMeters(0)
-        {
-        }
+        std::string Name = "";                          // Name of the meter
+        MeterType type = MeterType::Invalid;                       // type of meter
+        Constant::eResource resource = Constant::eResource::Invalid;              // Resource Type of the meter
+        SOVEndUseCat sovEndUseCat = SOVEndUseCat::Invalid;                        // End Use of the meter
+        std::string EndUseSub = "";                     // End Use subcategory of the meter
+        SOVGroup sovGroup = SOVGroup::Invalid;                         // Group of the meter
+        Constant::Units units = Constant::Units::Invalid;               // Units for the Meter
+        RT_IPUnits RT_forIPUnits; // Resource type number for IP Units (tabular) reporting
+
+        Real64 CurTSValue = 0.0;       // Current TimeStep Value (internal access)
+
+        std::array<MeterPeriod, (int)ReportFreq::Num> periods;    
+
+        MeterPeriod periodLastSM;
```

Comment:
These two metering periods are special. They don't correspond to a specific reporting frequency.